resilience4j-spring/src/main/java/io/github/resilience4j/timelimiter/configure/TimeLimiterAspect.java

```diff
@@ -139,8 +140,19 @@ private static TimeLimiter getTimeLimiterAnnotation(ProceedingJoinPoint proceedi
         }
     }
 
-    private static Object handleJoinPointCompletableFuture(
-            ProceedingJoinPoint proceedingJoinPoint, io.github.resilience4j.timelimiter.TimeLimiter timeLimiter) throws Throwable {
+    private static Object handleJoinPointFuture(
+            ProceedingJoinPoint proceedingJoinPoint, io.github.resilience4j.timelimiter.TimeLimiter timeLimiter, boolean blocking) throws Throwable {
+        if(blocking) {
+            return timeLimiter.executeFutureSupplier(() ->
+                    CompletableFuture.supplyAsync(() -> {
```

Comment:
The intercepted method must return a Future. That means the task is already executed in an ExecutorService.
The developer should be able use the ExecutorService of his choice.

The TimeLimiterAspect must not execute a method (task) implictly in it's own ExecutorService which is not configurable. The ThreadPoolBulkhead should be used to run tasks asynchronously.